google/cloud/documentai_toolbox/wrappers/page.py

```diff
@@ -33,28 +33,40 @@ class Table:
 
     Attributes:
         documentai_object (google.cloud.documentai.Document.Page.Table):
-            Required. The original google.cloud.documentai.Document.Page.Table object.
-        document_text (str):
-            Required. UTF-8 encoded text in reading order from the document.
+            Required. The original object.
         body_rows (List[List[str]]):
             Required. A list of body rows.
         header_rows (List[List[str]]):
             Required. A list of header rows.
     """
 
     documentai_object: documentai.Document.Page.Table = dataclasses.field(repr=False)
-    document_text: dataclasses.InitVar[str]
+    _page: "Page" = dataclasses.field(repr=False)
 
-    body_rows: List[List[str]] = dataclasses.field(init=False, repr=False)
-    header_rows: List[List[str]] = dataclasses.field(init=False, repr=False)
+    _body_rows: Optional[List[List[str]]] = dataclasses.field(
+        init=False, repr=False, default=None
+    )
+    _header_rows: Optional[List[List[str]]] = dataclasses.field(
+        init=False, repr=False, default=None
+    )
 
-    def __post_init__(self, document_text) -> None:
-        self.header_rows = _table_rows_from_documentai_table_rows(
-            table_rows=list(self.documentai_object.header_rows), text=document_text
-        )
-        self.body_rows = _table_rows_from_documentai_table_rows(
-            table_rows=list(self.documentai_object.body_rows), text=document_text
-        )
+    @property
+    def body_rows(self):
+        if self._body_rows is None:
```

Comment:
Looks like we are adding the same pattern in many places.  You might consider using this (which I think replaces the need for fields like `self._body_rows` here): https://docs.python.org/3/library/functools.html#functools.cached_property

Reply:
I agree, I thought that would be a better/cleaner option, but `@cached_property` allows for writes, which we don't want to allow on these properties.

From what I've found, this is the recommended procedure for implementing it.